mne_bids/mne_bids.py

```diff
@@ -424,60 +463,35 @@ def raw_to_bids(subject_id, task, raw_file, output_path, session_id=None,
 
     make_dataset_description(output_path, name=" ",
                              verbose=verbose)
-    _sidecar_json(raw, task, manufacturer, data_meta_fname, kind, verbose)
+    _sidecar_json(raw, task, manufacturer, data_meta_fname, kind,
+                  verbose)
     _channels_tsv(raw, channels_fname, verbose)
 
     events = _read_events(events_data, raw)
     if len(events) > 0:
         _events_tsv(events, raw, events_tsv_fname, event_id, verbose)
 
-    # for FIF, we need to re-save the file to fix the file pointer
-    # for files with multiple parts
-    if ext in ['.fif', '.gz']:
-        raw.save(raw_file_bids, overwrite=overwrite)
-    else:
-        if os.path.exists(raw_file_bids):
-            if overwrite:
-                os.remove(raw_file_bids)
-                sh.copyfile(raw_file, raw_file_bids)
-            else:
-                raise ValueError('"%s" already exists. Please set overwrite to'
-                                 ' True.' % raw_file_bids)
-
-    return output_path
+    # Writing of neural data
+    # Check if it is MEG (only writing to FIF)
+    # ----------------------------------------
+    if os.path.exists(raw_file_bids) and not overwrite:
+        raise ValueError('"%s" already exists. Please set'
+                         ' overwrite to True.' % raw_file_bids)
 
+    if verbose:
+        print('Writing data files to %s' % raw_file_bids)
 
-def _read_events(events_data, raw):
-    """Read in events data.
-
-    Parameters
-    ----------
-    events_data : str | array | None
-        The events file. If a string, a path to the events file. If an array,
-        the MNE events array (shape n_events, 3). If None, events will be
-        inferred from the stim channel using `find_events`.
-    raw : instance of Raw
-        The data as MNE-Python Raw object.
-
-    Returns
-    -------
-    events : array, shape = (n_events, 3)
-        The first column contains the event time in samples and the third
-        column contains the event id. The second column is ignored for now but
-        typically contains the value of the trigger channel either immediately
-        before the event or immediately after.
+    if ext in allowed_extensions_meg:
```

Comment:
you could generalize to use `ALLOWED_EXTENSIONS` since this will be used by all kinds

Reply:
this is already in preparation of the EEG if-else, which will go right below there. I just removed the EEG part in order to not merge it into 0.1.0 - that's why it might stick out a bit

Reply:
ok, that's fine